Diplom_1

Steamin01 opened the pull request.

Steamin01 committed:
UnitTests

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,109 @@
+    @Step("Создание настоящего бургера для теста операций")
+    private Burger createRealBurger() {
+        Bun bun = new Bun("test bun", 10f);
+        Ingredient ing1 = new Ingredient(IngredientType.FILLING, "cheese", 5f);
```
⛔️Нужно исправить.  Для тестирования используем моки, а не реальные объекты

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,109 @@
+
+    @Step("Проверка правильности расчета цены")
+    private void verifyPrice(float actual, float expected, Burger burger) {
+        assertEquals(expected, actual, 0.001f);
```
⛔️Нужно исправить. Для юнит-тестов применим подход: один тест, значит одна проверка. В этом тесте две проверки (Mockito.verify, assertEquals), а должна быть одна. Исправь, пожалуйста, этот момент во всем коде.

okryukova commented:
⛔️Нужно исправить. Нет отчета о покрытии. Покрытие для класса Burger должно быть 100%

Steamin01 committed:
Fix#1
- Добавлен Jacoco отчёт для класса Burger
- Внесены правки в BurgerTest
- Пересобран pom.xml для создания Jacoco отчёта

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,140 @@
+    }
+
+    @Parameterized.Parameters(name = "bunPrice, ingredientsPrices, expectedTotal")
+    public static Collection<Object[]> data() {
```
⛔️Нужно исправить.  В классе должны содержаться только параметризированные тесты. Если в тесте параметризация не нужна, вынеси его, пожалуйста, в другой класс

Steamin01 committed:
Fix#2
- Создан BurgerParametersTest
- Внесены правки в BurgerTest
